scalardl/src/scalardl/simple.clj

```diff
@@ -0,0 +1,124 @@
+  (:import (javax.json Json)
+           (java.util Optional)))
+
+(def ^:private ^:const CONTRACTS [{:name "read"
+                                   :class "com.scalar.jepsen.scalardl.Read"
+                                   :path "target/classes/com/scalar/jepsen/scalardl/Read.class"}
+                                  {:name "write"
+                                   :class "com.scalar.jepsen.scalardl.Write"
+                                   :path "target/classes/com/scalar/jepsen/scalardl/Write.class"}
+                                  {:name "cas"
+                                   :class "com.scalar.jepsen.scalardl.Cas"
+                                   :path "target/classes/com/scalar/jepsen/scalardl/Cas.class"}])
+
+(def ^:private ^:const NONCE "nonce")
+(def ^:private ^:const ASSET_KEY "key")
+(def ^:private ^:const ASSET_VALUE "value")
+(def ^:private ^:const ASSET_VALUE_NEW "new_value")
+
+(defn- create-argument
+  ([key]
+   (-> (Json/createObjectBuilder)
+       (.add ASSET_KEY key)
+       (.add NONCE (str (java.util.UUID/randomUUID)))
+       (.build)))
+  ([key value]
+   (-> (Json/createObjectBuilder)
+       (.add ASSET_KEY key)
+       (.add ASSET_VALUE value)
+       (.add NONCE (str (java.util.UUID/randomUUID)))
+       (.build)))
+  ([key cur next]
+   (-> (Json/createObjectBuilder)
+       (.add ASSET_KEY key)
+       (.add ASSET_VALUE cur)
+       (.add ASSET_VALUE_NEW next)
+       (.add NONCE (str (java.util.UUID/randomUUID)))
+       (.build))))
+
+(defrecord SimpleClient [initialized? client-service]
+  client/Client
+  (open! [_ test _]
+    (SimpleClient. initialized? (atom (dl/prepare-client-service test))))
+
+  (setup! [_ test]
+    (locking initialized?
+      (when (compare-and-set! initialized? false true)
+        (cassandra/create-tables test)
+        (Thread/sleep 10000)  ;; Wait for the table creation
+        (info "register a certificate and contracts")
+        (.registerCertificate @client-service)
+        (doseq [c CONTRACTS]
+          (.registerContract @client-service (:name c) (:class c) (:path c) (Optional/empty))))))
+
+  (invoke! [_ test op]
+    (case (:f op)
+      :read (let [argument (create-argument 1)
+                  resp (.executeContract @client-service "read" argument)]
+              (if (util/success? resp)
+                (assoc op :type :ok :value (-> resp
+                                               util/response->obj
+                                               (.getInt ASSET_VALUE)))
+                (do
+                  (reset! client-service
+                          (dl/try-switch-server! @client-service test))
+                  (assoc op :type :fail :error (.getMessage resp)))))
+
+      :write (let [v (:value op)
+                   argument (create-argument 1 v)
+                   resp (.executeContract @client-service "write" argument)]
+               (if (or (util/success? resp)
+                       (and (util/unknown? resp)
```

Comment:
As we discussed in person, maybe we can throw `info` after some retries failed ?

Reply:
Thank you for your proposal!
Changed the handling the unknown status.